api/pkg/transformer/pipeline/compiler.go

```diff
@@ -245,6 +245,9 @@ func (c *Compiler) doCompilePipeline(pipeline *spec.Pipeline, pipelineType types
 }
 
 func (c *Compiler) parseUpiPreprocessOutput(outputSpec *spec.UPIPreprocessOutput) (Op, error) {
+	if outputSpec.PredictionTableName == "" && len(outputSpec.TransformerInputTableNames) == 0 {
+		return nil, fmt.Errorf(`"predictionTableName" or "transformerInputTableNames" must be set for upi preprocess output spec`)
```

Comment:
The error message should be using `and` instead of `or`:

```
return nil, fmt.Errorf(`"predictionTableName" and "transformerInputTableNames" must be set for upi preprocess output spec`)
```

Another option is to separate the validation to make it more verbose and clearer:

```
if outputSpec.PredictionTableName == "" {
		return nil, fmt.Errorf(`"predictionTableName" must be set for upi preprocess output spec`)
}

if len(outputSpec.TransformerInputTableNames) == 0 {
		return nil, fmt.Errorf(`"transformerInputTableNames" must be set for upi preprocess output spec`)
```

Reply:
`predictionTableName` and `transformerInputTableNames` are not necessarily set, at least one of them must be set.